src/platform/forms-system/test/js/widgets/RadioWidget.unit.spec.jsx

```diff
@@ -161,35 +174,66 @@ describe('Schemaform <RadioWidget>', () => {
-    expect(inputsFirstSelected[1].props['data-selected']).to.be.undefined;
-    expect(inputsFirstSelected[2].props['data-test']).to.equal('third');
-    expect(inputsFirstSelected[2].props['data-selected']).to.be.undefined;
+
+    await fireEvent.click($('input[value="1"]', container));
+    await waitFor(() => {
+      expect(getAttr(container, 0, 'data-test')).to.equal('first');
+      expect(getAttr(container, 0, 'data-selected')).to.equal('first_1');
+      expect(getAttr(container, 1, 'data-test')).to.equal('second');
+      expect(getAttr(container, 1, 'data-selected')).to.be.null;
+      expect(getAttr(container, 2, 'data-test')).to.equal('third');
+      expect(getAttr(container, 2, 'data-selected')).to.be.null;
+    });
 
     // second option selected
-    tree.reRender({ value: '2', onChange, options });
-    const inputsSecondSelected = tree.everySubTree('input');
-    expect(inputsSecondSelected[0].props['data-test']).to.equal('first');
-    expect(inputsSecondSelected[0].props['data-selected']).to.be.undefined;
-    expect(inputsSecondSelected[1].props['data-test']).to.equal('second');
-    expect(inputsSecondSelected[1].props['data-selected']).to.equal('second_2');
-    expect(inputsSecondSelected[2].props['data-test']).to.equal('third');
-    expect(inputsSecondSelected[2].props['data-selected']).to.be.undefined;
+    rerender(<RadioWidget value="2" onChange={onChange} options={options} />);
+    await waitFor(() => {
+      expect(getAttr(container, 0, 'data-test')).to.equal('first');
+      expect(getAttr(container, 0, 'data-selected')).to.be.null;
+      expect(getAttr(container, 1, 'data-test')).to.equal('second');
+      expect(getAttr(container, 1, 'data-selected')).to.equal('second_2');
+      expect(getAttr(container, 2, 'data-test')).to.equal('third');
+      expect(getAttr(container, 2, 'data-selected')).to.be.null;
+    });
 
     // third option selected
-    tree.reRender({ value: '3', onChange, options });
-    const inputsThirdSelected = tree.everySubTree('input');
-    expect(inputsThirdSelected[0].props['data-test']).to.equal('first');
-    expect(inputsThirdSelected[0].props['data-selected']).to.be.undefined;
-    expect(inputsThirdSelected[1].props['data-test']).to.equal('second');
-    expect(inputsThirdSelected[1].props['data-selected']).to.be.undefined;
-    expect(inputsThirdSelected[2].props['data-test']).to.equal('third');
-    expect(inputsThirdSelected[2].props['data-selected']).to.equal('third_3');
+    rerender(<RadioWidget value="3" onChange={onChange} options={options} />);
+
+    await waitFor(() => {
+      expect(getAttr(container, 0, 'data-test')).to.equal('first');
+      expect(getAttr(container, 0, 'data-selected')).to.be.null;
+      expect(getAttr(container, 1, 'data-test')).to.equal('second');
+      expect(getAttr(container, 1, 'data-selected')).to.be.null;
+      expect(getAttr(container, 2, 'data-test')).to.equal('third');
```

Comment:
And this one covers the 3rd one selected because it's a radio widget so only one can be selected at the same time.